File: 1768-merge-strings-alternately/1768-merge-strings-alternately.go
func mergeAlternately(word1 string, word2 string) string {
        len1 := len(word1)
        len2 := len(word2)
        result := make([]byte, 0, len1+len2)

        i := 0

        for ; i < len1 && i < len2; i++ {
            result = append(result, word1[i])
            result = append(result, word2[i])
        }

        if i < len1 {
            result = append(result, word1[i:]...)
        } else if i < len2 {
            result = append(result, word2[i:]...)
        }
        return string(result)
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.swift
class Solution {
    func mergeAlternately(_ word1: String, _ word2: String) -> String {
        
        let chars1 = Array(word1)
        let chars2 = Array(word2)
        
        var merged = ""
        let maxLength = max(chars1.count, chars2.count)
        
        for i in 0..<maxLength {
            if i < chars1.count {
                merged.append(chars1[i])
            }
            if i < chars2.count {
                merged.append(chars2[i])
            }
        }
        return merged
        
    }
}

